scripts/token-mold.js

```diff
@@ -348,8 +348,9 @@ export default class TokenMold {
   _modifyName(data, actor, sceneId) {
     let name = actor.data.token.name;
 
-    if (this.data.name.replace !== "" && this.data.name.replace !== "nothing")
+    if (["remove", "replace"].includes(this.data.name.replace) && !(this.data.name.baseNameOverride && event.getModifierState("Shift"))) {
```

Comment:
I modified the logic here to be a bit more clear. Name should only be cleared out if the options chosen are "remove" or "replace".
If the setting is enabled in the options, holding shift while dragging a token to the canvas will not remove the actors name. This is useful in the event that you know your group has previously identified a type of creature.